### DISCUSSION TITLE
How to open google search box in app window in Avalonia UI?

### DISCUSSION THREAD

**Discussion by al-zakaria** [AvaloniaUI/Avalonia#15743] [Q&A]
I have a TextBox and when I click a button, I want to make it so that a window opens with search results for the query that I entered into the textbox. how to do it?

**Comment by maxkatz6** [comment]
Can you give an example of such search box? Do you mean just opening a google page with search query?

**Reply by al-zakaria** [reply]
if you are talking about an existing such application, then you can give an example of the FireFox browser. There is a text field on the initial screen and when we enter something there, the Google search engine simply opens in the same window. that is, I'm talking about doing it like in FireFox

**Reply by thevortexcloud** [reply]
Do you want to open any web browser or a specific one? If you don't care which one then it's fairly simple, if you need a specific one it gets far more complicated. You need to work out where things are installed at that point (which will vary by platform) and they can be installed pretty much anywhere (at least on Linux they will generally be on the path, whereas on Windows you need to read registry keys to find the installation location). They may also not even be registered with the OS but that is a situation most people don't handle (or if they do, they require people to manually select the installation location).

**Comment by silent21century** [comment]
this will open it on the default browser, im not aware of a built in way to doing it tho 
```cs 
var query = TextBox.Text.Replace(" ","+")
ProcessStartInfo psi = new ProcessStartInfo("cmd", $"/c start https://www.google.com/search?q={query}")
{
    CreateNoWindow = true
};

Process.Start(psi);
        
```

**Reply by thevortexcloud** [reply]
Avalonia now has a built in implementation that will do this for you on desktops, not sure if it's been released yet though.

https://github.com/AvaloniaUI/Avalonia/pull/14320

You also don't need to launch cmd (which is a windows only thing anyway). You can just pass it the URL and it will work out the rest.

Also your input sanitisation likely wont work. I am not sure how Google's search parameters work but you likely need to actually URL encode the query otherwise if something inputs something like "bed & breakfast" the search wont work properly.

**Reply by al-zakaria** [reply]
no, I'm saying not to open a new default browser window, but to launch the Google search engine in my application window.

**Reply by rabbitism** [reply]
> no, I'm saying not to open a new default browser window, but to launch the Google search engine in my application window.

Are you trying to create a browser?

**Reply by al-zakaria** [reply]
yes, I want to try to do something like this for practice

**Reply by thevortexcloud** [reply]
Assuming you don't want to create your own browser engine (that's a very complicated thing to do), you can either embed a real web browser inside your app or use the HTML renderer.

https://github.com/AvaloniaUI/Avalonia.HtmlRenderer

https://github.com/OutSystems/CefGlue
